pkg/asset/machines/master.go

```diff
@@ -114,9 +137,110 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 	default:
 		return fmt.Errorf("invalid Platform")
 	}
+
+	machines := []interface{}{}
+	if len(m.Machines) == 0 {
+		for _, machine := range m.MachinesDeprecated {
+			machines = append(machines, machine)
+		}
+	} else {
+		for _, machine := range m.Machines {
+			machines = append(machines, machine)
+		}
+	}
+
+	for i, machine := range machines {
+		data, err := yaml.Marshal(machine)
+		if err != nil {
+			return errors.Wrapf(err, "marshal master %d", i)
+		}
+
+		m.FileList = append(m.FileList, &asset.File{
+			Filename: filepath.Join(directory, fmt.Sprintf(MasterMachineFileName, i)),
+			Data:     data,
+		})
+	}
+
+	asset.SortFiles(m.FileList)
+
 	return nil
 }
 
+// Files returns the files generated by the asset.
+func (m *Master) Files() []*asset.File {
+	return m.FileList
+}
+
+// Load reads the asset files from disk.
+func (m *Master) Load(f asset.FileFetcher) (found bool, err error) {
+	file, err := f.FetchByName(filepath.Join(directory, MasterUserDataFileName))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	m.FileList = []*asset.File{file}
+
+	m.Machines = []machineapi.Machine{}
+	m.MachinesDeprecated = []clusterapi.Machine{}
+	fileList, err := f.FetchByPattern(filepath.Join(directory, fmt.Sprintf(MasterMachineFileName, "*")))
+	if err != nil {
+		return true, err
+	}
+	m.FileList = append(m.FileList, fileList...)
+	for _, file := range fileList {
+		machine := &machineapi.Machine{}
+		err = yaml.Unmarshal(file.Data, &machine)
+		if err != nil {
+			return true, errors.Wrapf(err, "unmarshal %q", file.Filename)
+		}
+
+		rawProvider := machine.Spec.ProviderSpec.Value.Raw
+		providerType := &runtime.TypeMeta{}
+		err = yaml.Unmarshal(rawProvider, &providerType)
+		if err != nil {
+			return true, errors.Wrapf(err, "unmarshal %q provider type", file.Filename)
+		}
+
+		switch providerType.APIVersion {
+		case "awsproviderconfig.k8s.io/v1alpha1":
+			provider := &awsprovider.AWSMachineProviderConfig{}
+			err = yaml.Unmarshal(rawProvider, &provider)
+			if err != nil {
+				return true, errors.Wrapf(err, "unmarshal %s AWS provider", file.Filename)
+			}
+			machine.Spec.ProviderSpec.Value = &runtime.RawExtension{Object: provider}
+			m.Machines = append(m.Machines, *machine)
+		case "libvirtproviderconfig.k8s.io/v1alpha1":
+			provider := &libvirtprovider.LibvirtMachineProviderConfig{}
+			err = yaml.Unmarshal(rawProvider, &provider)
+			if err != nil {
+				return true, errors.Wrapf(err, "unmarshal %s libvirt provider", file.Filename)
+			}
+			m.Machines = append(m.Machines, *machine)
+		case "openstack.cluster.k8s.io/v1alpha1":
```

Comment:
This should be `openstackproviderconfig.k8s.io/v1alpha1`.

Reply:
> This should be `openstackproviderconfig.k8s.io/v1alpha1`.

Oops, thanks :).  Fixed in 24aa42f -> 955cf95.